src/command/defaults/KillCommand.php

```diff
@@ -53,7 +54,7 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
 			return true;
 		}
 
-		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, 1000));
+		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, PHP_FLOAT_MAX));
```

Comment:
works?
```suggestion
		$player->setHealth(0);
		$player->attack(new EntityDamageEvent($player, EntityDamageEvent::CAUSE_SUICIDE, 1000));
```

Reply:
It wouldn't make sense if the damage event gets called after the entity died. And might as well put the damage as `0`. SOFe's idea seems better. Just use `INF` damage

Reply:
What do you think about how PowerNukkit does it
https://github.com/PowerNukkit/PowerNukkit/blob/master/src/main/java/cn/nukkit/command/defaults/KillCommand.java#L50-L56

Reply:
Infinite damage can do just that

Reply:
This change will cause the command to be effective against creative players which is not intended.

Reply:
/kill already works on Creative players in PM (which I don't intend to change, as it's useful for returning to the spawnpoint). However, setHealth(0) calls kill(), which would prevent plugins from cancelling the event.